src/main/java/org/folio/linked/data/service/resource/AssignAuthorityTarget.java

```diff
@@ -0,0 +1,45 @@
+package org.folio.linked.data.service.resource;
+
+import static java.lang.Boolean.FALSE;
+import static java.util.Map.of;
+import static org.apache.commons.lang3.ObjectUtils.notEqual;
+import static org.folio.ld.dictionary.ResourceTypeDictionary.CONCEPT;
+import static org.folio.ld.dictionary.ResourceTypeDictionary.FAMILY;
+import static org.folio.ld.dictionary.ResourceTypeDictionary.JURISDICTION;
+import static org.folio.ld.dictionary.ResourceTypeDictionary.MEETING;
+import static org.folio.ld.dictionary.ResourceTypeDictionary.ORGANIZATION;
+import static org.folio.ld.dictionary.ResourceTypeDictionary.PERSON;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.Set;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import org.apache.commons.collections4.CollectionUtils;
+import org.folio.ld.dictionary.ResourceTypeDictionary;
+
+@Getter
+@RequiredArgsConstructor
+public enum AssignAuthorityTarget {
+
+  CREATOR_OF_WORK(of(CONCEPT, Set.of(PERSON, FAMILY, ORGANIZATION, JURISDICTION, MEETING)));
+
+  private final Map<ResourceTypeDictionary, Set<ResourceTypeDictionary>> combinedTypes;
```

Comment:
Hi @AndreiBordak, It is not very clear what the key & value of this Map represent.

Reply:
Reworked solution.